homeassistant/components/dwd_weather_warnings/util.py

```diff
@@ -0,0 +1,32 @@
+"""Util functions for the dwd_weather_warnings integration."""
+
+from __future__ import annotations
+
+from homeassistant.core import HomeAssistant
+
+from .const import LOGGER
+
+
+def get_position_data(
+    hass: HomeAssistant, device_tracker: str
+) -> tuple[float, float] | None:
+    """Extract longitude and latitude from a device tracker."""
+    entity = hass.states.get(device_tracker)
+    if entity is None:
+        return None
```

Comment:
Raise a custom exception instead for all the cases that return None here. That's not expected and the function caller needs to handle that.

Reply:
What's the preferred way to handle the exceptions in a coordinator? For the config flow I'd guess an error for the respective exceptions is good but I'm not sure for the coordinator.

Currently, it would set the position to `None` which would still instantiate the API but it would probably become unavailable

Reply:
Probably create a repair issue and raise `UpdateFailed` if it's during a data refresh.

Reply:
I think this would be better in a follow-up PR, too. I've started implementing this and in my opinion it's too much for this scope. A basic repair issue is easy enough for all the places that call this function but it's very user friendly to just have an issue. I'd like to have the integration automatically remove the issue, if the problem was fixed and when appropriate offer a repair flow too.

But this doesn't have anything to do with this PR

Reply:
Yes, that's ok for now.

Reply:
Updated in commit 1bec061985b3573c7b3089ea62d17edda8422a4a